src/core/routing/route-pruning.ts

```
@@ -0,0 +1,554 @@
+  efficiencyScore: number;
+  preserved: boolean;
+}
+
+/**
+ * Apply intelligent pruning to routes to remove inefficient or redundant options
+ */
+export function pruneRoutes(
+  routes: Route[],
+  graph: TransitGraph,
+  destinationIdOrMaxRoutes?: string | number,
+  maxRoutes: number = 5
```

Biome (1.9.4)

[error] This type annotation is trivially inferred from its initialization.

Safe fix: Remove the type annotation.


(lint/style/noInferrableTypes)